WordPress/Classes/Models/PublicizeService+Lookup.swift

```diff
@@ -0,0 +1,31 @@
+extension PublicizeService {
+    /// Finds a cached `PublicizeService` matching the specified service name.
+    ///
+    /// - Parameter name: The name of the service. This is the `serviceID` attribute for a `PublicizeService` object.
+    ///
+    /// - Returns: The requested `PublicizeService` or nil.
+    ///
+    static func lookupPublicizeServiceNamed(_ name: String, in context: NSManagedObjectContext) throws -> PublicizeService? {
+        let request = NSFetchRequest<PublicizeService>(entityName: PublicizeService.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "serviceID = %@", name)
+        return try context.fetch(request).first
+    }
+
+    @objc(lookupPublicizeServiceNamed:inContext:)
+    static func objc_lookupPublicizeServiceNamed(_ name: String, in context: NSManagedObjectContext) -> PublicizeService? {
+        try? lookupPublicizeServiceNamed(name, in: context)
```

Comment:
LOL. I just commented above about the code being better because it doesn't return `.none` as a sign of error... 🙃 

Of course, this is a special case method, behaving that way to fit with the Objective-C layer 😄

Reply:
Yeah, a little compromise for Objective-C code.